staging/src/k8s.io/api/bulk/v1alpha1/types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkRequest contains a single request to bulk api.
+type BulkRequest struct {
```

Comment:
ChannelRequest? Some number of these are sent after a long running request (channel) is opened, right?

(Maybe we want a bulk POST in the future. The current name seems overly broad.)

Reply:
I think Bulk is a fine name for the group, but the resource / operation maybe should be "multiplexed" or something like that? When I think of "bulk" I think of a request body with N chunks of work for the server to do.

Reply:
>> Maybe we want a bulk POST in the future. The current name seems overly broad.
Actually bulk POST is already implemented :) 
It is just ~130 additional lines with current architecture (see bulk/http.go), so it doesn't make a lot of sense to not implement it from start.